### DISCUSSION TITLE
Add physics to Instance component

### DISCUSSION THREAD

**Discussion by netgfx** [pmndrs/drei#807] [Q&A] [answered]
How would one add physics (cannon) to an `Instance` ? Tried with `[ref] = useBox()` but that did nothing.

**Comment by mwmwmw** [comment, marked as answer]
Some example code would help sort out your issue. 

If you're using `@react-three/cannon`, don't forget to wrap your physics objects in the `<Physics>` component.

Here is a working example https://codesandbox.io/s/usecannon-physics-basic-example-y2t7tu?file=/src/index.js

**Reply by Dan0net** [reply]
Unfortunately I don't think this is the answer, as the code example doesn't use `<Instances>` or `<Instance>`.

It's certainly possible to use `@react-three/cannon` with an `instancedMesh` as in this example: https://codesandbox.io/s/r3f-cannon-instanced-physics-g1s88?file=/src/index.js

However, I'm also struggling to find an examples where something like `useBox` has been integrated into an  `<Instances>` or `<Instance>` component. Are there any other ideas?

**Comment by drcmda** [comment]
it doesnt really make sense to mix drei/instance and physics. the whole point of drei/instance is that you are able to control the instances position, but when you couple an object with physics then the physics-engine controls position. react-three/cannon can handle instances out of the box without you having to do anything.

**Comment by iaruso** [comment]
I have a problem with this, it does works as intended using useSphere(), but when using useBox() it only collides with usePlane() object but not useTrimesh() ones. It's weird.
